transformers/events/DOCUMENTATION.md

```diff
@@ -2,116 +2,206 @@
+       will be the package your transformer is stored in.
+    1. Create three files - `transformer.go`, `transformer_test.go` and
+       `<package_name>_test.go` in the new directory. See the other transformers
+       for examples.
+    1. `<package_name>_test.go` lets `gingko` run tests in this directory. The
+       file should look something like this:
+       
+```go
+ package <package_name>_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestLogMinSell(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "<PackageName>  Suite")
+}
+```
+   1. Implement the toModels function in `transformer.go` with the appropriate
+   testing in `transformer_test.go` of course. This means converting the raw log 
+   into a go struct. 
+   1. For custom events you can convert each EventLog entry into an entity
+   using the function `contract.UnpackLog(&entity, "<EventLogName>", log.Log)`.
+   The entity is usually defined (by you) in a file called `entity.go` in the
+   same package as the transformer. See the transformer in
+   `transformers/events/log_make` for an example of this method.
+   1. After converting the log entry to an entity, look up all of it's foreign
+      keys from it's data and assign them to the entity.
+   1. From the entity create an InsertionModel, which you return.
+   1. For `LogNote` events you'll need to look at the method signature of the
+   method that is calling them, because LogNote events are a generic structure.
+   For example:
+       * the `tend` method is called on the [flip.sol
+       contract](https://github.com/makerdao/dss/blob/master/src/flip.sol#L123), 
+       and its method signature looks like this: `tend(uint,uint,uint)`.
+       * The first four bytes of the Keccak-256 hashed method signature will be 
+       located in `topic[0]` on the log.
+       * The message sender will be in `topic[1]`.
+       * The first parameter passed to `tend` becomes `topic[2]`.
+       * The second parameter passed to `tend` will be `topic[3]`.
+       * Any additional parameters will be in the log's data field.
+       * More detail is located in the [DSNote
+         repo](https://github.com/dapphub/ds-note).
+       * For an example implementation look at the `flip_sol` transformer.
+   1. If, while implementing the transformer and its corresponding unit tests
+      you find you need to migrate the database see below. 
+
+### Store the data in the database
+1. Use the `make new_migration` task to create a new migration.
+    * Each event log has its own table in the database.
+    * The specific log event tables are all created in the `maker` schema.
+1. The new migration can be run by running `make test` or `make migration NAME=vulcanize_test`.
+1. To verify that the migration will work create an integration for the shiny
+   new transformer in [`integration_tests`](../integration_tests).
+1. Integration tests can be run with `make integrationtest`.
```

Comment:
I sorta shudder at the thought of describing it in detail because it's something we need to streamline, but maybe worth covering something about what goes into writing an integration test for an event transformer?

Reply:
Probably - I think I was running into some documentation fatigue at this point.